tests/template_tests/test_loaders.py

```diff
@@ -176,7 +176,18 @@ def test_not_installed(self):
 class FileSystemLoaderTests(SimpleTestCase):
 
     def setUp(self):
-        self.engine = Engine()
+        self.engine = Engine(dirs=[TEMPLATE_DIR])
+
+    @staticmethod
+    @contextmanager
+    def set_permissions(path, mask):
+        old = oct(os.stat(path).st_mode)
+        new = "%s%s" % (old[:-3], mask)
+        try:
+            os.chmod(path, int(new, base=8))
```

Comment:
`os.chmod` should be before the `try`. It doesn't make a big difference but that's the common pattern in general.